src/main/java/io/reactivex/subjects/UnicastSubject.java

```diff
@@ -102,37 +105,96 @@
+     * non-overlapped with any active replay.
+     *
+     * @param <T> the value type
+     * @param capacityHint the hint to size the internal unbounded buffer
+     * @param onTerminate the callback to run when the Subject is terminated or cancelled, null not allowed
+     * @param delayError deliver pending onNext events before onError
+     * @return an UnicastSubject instance
+     */
+    @CheckReturnValue
+    public static <T> UnicastSubject<T> create(int capacityHint, Runnable onTerminate, boolean delayError) {
+        return new UnicastSubject<T>(capacityHint, onTerminate, delayError);
+    }
+
+    /**
+     * Creates an UnicastSubject with an internal buffer capacity hint 16 and given delay error flag
+     *
+     * <p>The callback, if not null, is called exactly once and
+     * non-overlapped with any active replay.
+     *
+     * @param <T> the value type
+     * @param delayError deliver pending onNext events before onError
+     * @return an UnicastSubject instance
+     */
+    @CheckReturnValue
+    public static <T> UnicastSubject<T> create(boolean delayError) {
+        return new UnicastSubject<T>(bufferSize(), delayError);
+    }
+
+
+    /**
+     * Creates an UnicastSubject with the given capacity hint and delay error flag.
      * @param capacityHint the capacity hint for the internal, unbounded queue
+     * @param delayError deliver pending onNext events before onError
      * @since 2.0
      */
-    UnicastSubject(int capacityHint) {
+    UnicastSubject(int capacityHint, boolean delayError) {
         this.queue = new SpscLinkedArrayQueue<T>(ObjectHelper.verifyPositive(capacityHint, "capacityHint"));
         this.onTerminate = new AtomicReference<Runnable>();
+        this.delayError = delayError;
         this.actual = new AtomicReference<Observer<? super T>>();
         this.once = new AtomicBoolean();
         this.wip = new UnicastQueueDisposable();
     }
 
     /**
-     * Creates an UnicastProcessor with the given capacity hint and callback
-     * for when the Processor is terminated normally or its single Subscriber cancels.
+     * Creates an UnicastSubject with the given capacity hint and callback
+     * for when the Subject is terminated normally or its single Subscriber cancels.
      * @param capacityHint the capacity hint for the internal, unbounded queue
-     * @param onTerminate the callback to run when the Processor is terminated or cancelled, null not allowed
+     * @param onTerminate the callback to run when the Subject is terminated or cancelled, null not allowed
      * @since 2.0
-     */
+     *
+     * */
     UnicastSubject(int capacityHint, Runnable onTerminate) {
+        this(capacityHint, onTerminate, true);
+    }
+
+    /**
+     * Creates an UnicastSubject with the given capacity hint.
+     * @param capacityHint the capacity hint for the internal, unbounded queue
+     * @since 2.0
+     */
+    UnicastSubject(int capacityHint) {
```

Comment:
Is this one really needed? The factories could just go and call `UnicastSubject(capacityHint, true)` ?